# <u><b>Final Project: Student Loan Dataset</b></u>
## Milestone 2
## By Paulina Mostek
## <u>Objective:</u> Explore the National Student Loan Data System dataset using Python, and perform data cleaning using pandas.
### Datasets used: https://catalog.data.gov/dataset/national-student-loan-data-system-722b0
* 1617FedSchoolCodeList.xlsx (Federal School Code List) - This may be needed to crossreference school data
* FL_Dashboard_AY_2009_2010_Q1.xls (Q1 of 2009-2010 School Year)
### We will be exploring the original Excel files and converted CSV files in order to determine which file type should be used for the final project.

# <u>Data Exploration: Why this data?</u>
#### In this project, we will be analyzing the data of student loans by a varying number of factors such as school type, state, and type of loan. 
#### Clients: The U.S. Department of Education or higher education policy analysts may benefit from analysis of this data. 
#### The analysis will....
* identify loan distribution patterns by institution type (public vs private) and state.
* help target policy changes to address student debt concentration.
* inform funding allocation for community/technical colleges vs universities.
* assist in developing guidelines for financial aid reform or loan caps.
* highlight trends that will identify priorities for educational investment.
#### Additionally, if the findings of these data queries are published publicly, <u>students</u> may be a potential client for this analysis. The analysis could aid students’ decisions on what states to attend school and what universities to attend. Students may be more inclined to apply to universities with a higher loan disbursement rate.


# <u>Data Exploration: Questions</u>
### We will be asking the following questions of the data and plotting the results:
1. Do community colleges or technical colleges originate more loans overall?
2. Which states have the highest total loan disbursements?
3. Do private or public schools disburse more in federal student loans?
4. Do institutions with “college” in the name receive more loans than those with “university”?
5. Are unsubsidized loans more common at private colleges than public ones?
### For relevancy, we will only explore data concerning schools in the US. Foreign schools will be excluded from our analysis.

# <u>Data Cleaning</u>
### After observing the data in Excel, potential issues have been identified.
### Possible quality check issues (for data cleaning):
* Additional school types - "PROPRIETARY" may require deletion if containing empty values or comparing private vs public schools.
* Additional school types - "FOREIGN PUBLIC, FOREIGN PRIVATE" are displayed throughout the data, which need to be deleted as we are only investigating US schools.
* Missing rows for school code and financial information: values are "-" or "0" instead of numbers.
* Other columns may need to be investigated for missing rows.
* Several Rows in the "Award Year Summary" Table are missing, particularly for parent and grad loans.


# <u>Import libraries</u>
## The following libraries will be imported for our code:
* csv: For reading xls files as csv.
* pandas: For loading data into dataframes.
* numpy: For handling arrays.
* matplotlib and seaborns: For plotting data observations.
* statistics: For exploratory analysis and checking basic information of the table.
* xlrd: This library must be installed (pip install xlrd) AND imported to read XLS files.

In [1]:
import csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statistics as stats
import seaborn as sns
import xlrd

# <u>Initial Exploratory Analysis</u>
### We will import the data into a pandas dataframe using the appropriate Python library for the dataset file type.
### We will explore basic properties of each dataset such as...
* describe()
* columns
* shape
* dtypes
* head(), tail(), sample()
* info()

### Load in each dataset into a pandas dataframe
* The original dataset is an Excel file containing multiple sheets.
* We will be loading the first two sheets into a dictionary.
* We will also be using a CSV version of the same file.
* We will compare the CSV and Excel files to decide which file to use for data exploration.

### Step 1: Load both original XLS files and CSV Files
* Use a dictionary to load in the first two sheets: "Quarterly Activity" and "Award Year Summary"
* Access each sheet via key (sheet name)
* Load in the CSV files, which are saved separately.

In [2]:
#XLS APPROACH
#Use a dictionary to load in multiple sheets
#Load two specific sheets by name
sheets = pd.read_excel("FL_Dashboard_AY2009_2010_Q1.xls", sheet_name=["Quarterly Activity", "Award Year Summary"])

#Access each sheet (excel)
qa = sheets["Quarterly Activity"]
ays = sheets["Award Year Summary"]

#CSV APPROACH
#Load the CSV versions
qa_csv = pd.read_csv("FL_Dashboard_AY2009_2010_Q1 - Quarterly Activity.csv")
ays_csv = pd.read_csv("FL_Dashboard_AY2009_2010_Q1 - Award Year Summary.csv")

### Step 2: Compare XLS and CSV Files 
* Check the shape of both: they are identical
* Compare column names of both: they are identical in column name, number, and object type
* Compare full dataframes: Missing values

#### Shape

In [4]:
#Quarterly Activity
print("Excel QA shape:", qa.shape)
print("CSV QA shape:", qa_csv.shape)

#Award Year Summary
print("Excel AYS shape:", ays.shape)
print("CSV AYS shape:", ays_csv.shape)

Excel qa shape: (3825, 25)
CSV qa shape: (3825, 25)
Excel ays shape: (3825, 25)
CSV ays shape: (3825, 25)


#### Columns

In [10]:
#Quarterly Activity
print("QA column names:")
print("Excel:",qa.columns)
print("CSV:",qa_csv.columns)

#Award Year Summary
print("AYS column names:")
print("Excel:",ays.columns)
print("CSV:",ays_csv.columns)


QA column names:
Excel: Index(['2009-2010 Award Year FFEL Volume by School', 'Unnamed: 1',
       'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6',
       'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11',
       'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16', 'Unnamed: 17', 'Unnamed: 18', 'Unnamed: 19',
       'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23',
       'Unnamed: 24'],
      dtype='object')
CSV: Index(['2009-2010 Award Year FFEL Volume by School', 'Unnamed: 1',
       'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6',
       'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11',
       'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16', 'Unnamed: 17', 'Unnamed: 18', 'Unnamed: 19',
       'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23',
       'Unnamed: 24'],
      dtype='object')
AYS column names:
Excel: Index(['

#### Compare full dataframes

In [11]:
#Quarterly Analysis
comparison_qa = qa.compare(qa_csv, keep_shape=True, keep_equal=False)
print("Differences in QA:")
print(comparison_qa)


Differences in QA:
     2009-2010 Award Year FFEL Volume by School       Unnamed: 1        \
                                           self other       self other   
0                                           NaN   NaN        NaN   NaN   
1                                           NaN   NaN        NaN   NaN   
2                                           NaN   NaN        NaN   NaN   
3                                           NaN   NaN        NaN   NaN   
4                                           NaN   NaN        NaN   NaN   
...                                         ...   ...        ...   ...   
3820                                        NaN   NaN        NaN   NaN   
3821                                        NaN   NaN        NaN   NaN   
3822                                        NaN   NaN        NaN   NaN   
3823                                        NaN   NaN        NaN   NaN   
3824                                        NaN   NaN        NaN   NaN   

     Unnamed: 2   

In [12]:
#Award Year Summary
comparison_ays = ays.compare(ays_csv, keep_shape=True, keep_equal=False)
print("Differences in AYS:")
print(comparison_ays)

Differences in AYS:
     2009-2010 Award Year FFEL Volume by School       Unnamed: 1        \
                                           self other       self other   
0                                           NaN   NaN        NaN   NaN   
1                                           NaN   NaN        NaN   NaN   
2                                           NaN   NaN        NaN   NaN   
3                                           NaN   NaN        NaN   NaN   
4                                           NaN   NaN        NaN   NaN   
...                                         ...   ...        ...   ...   
3820                                        NaN   NaN        NaN   NaN   
3821                                        NaN   NaN        NaN   NaN   
3822                                        NaN   NaN        NaN   NaN   
3823                                        NaN   NaN        NaN   NaN   
3824                                        NaN   NaN        NaN   NaN   

     Unnamed: 2  

#### Observation: Missing values
* Since there are missing values in the comparison, we will try to detect which file (csv or xls) has the missing values.

## Step 3: Identify Missing values
* View columns with missing values in both dataframes
* View rows with missing values in both dataframes
* Count missing values in each dataframe (CSV and XLS) to determine which file type to use

In [13]:
#View columns
#Quarterly Analysis
#For Excel version
print("Missing values in Excel (qa):")
print(qa.isna().sum())

#For CSV version
print("Missing values in CSV (qa_csv):")
print(qa_csv.isna().sum())

Missing values in Excel (qa):
2009-2010 Award Year FFEL Volume by School     1
Unnamed: 1                                     4
Unnamed: 2                                     4
Unnamed: 3                                    22
Unnamed: 4                                     4
Unnamed: 5                                     3
Unnamed: 6                                     4
Unnamed: 7                                     4
Unnamed: 8                                     4
Unnamed: 9                                     4
Unnamed: 10                                    3
Unnamed: 11                                    4
Unnamed: 12                                    4
Unnamed: 13                                    4
Unnamed: 14                                    4
Unnamed: 15                                    3
Unnamed: 16                                    4
Unnamed: 17                                    4
Unnamed: 18                                    4
Unnamed: 19                            

In [14]:
#Award Year Summary
#For Excel version
print("Missing values in Excel (ays):")
print(ays.isna().sum())

#For CSV version
print("Missing values in CSV (ays_csv):")
print(ays_csv.isna().sum())

Missing values in Excel (ays):
2009-2010 Award Year FFEL Volume by School     1
Unnamed: 1                                     4
Unnamed: 2                                     4
Unnamed: 3                                    22
Unnamed: 4                                     4
Unnamed: 5                                     3
Unnamed: 6                                     4
Unnamed: 7                                     4
Unnamed: 8                                     4
Unnamed: 9                                     4
Unnamed: 10                                    3
Unnamed: 11                                    4
Unnamed: 12                                    4
Unnamed: 13                                    4
Unnamed: 14                                    4
Unnamed: 15                                    3
Unnamed: 16                                    4
Unnamed: 17                                    4
Unnamed: 18                                    4
Unnamed: 19                           

#### Identify missing rows

In [15]:
#Identify missing rows

#Quarterly Analysis
print("Quarterly Analysis")
print("Rows with missing values in Excel:")
print(qa[qa.isna().any(axis=1)])

print("Rows with missing values in CSV:")
print(qa_csv[qa_csv.isna().any(axis=1)])


Quarterly Analysis
Rows with missing values in Excel:
             2009-2010 Award Year FFEL Volume by School  \
0     Award Year Quarterly Activity  (07/01/2009-09/...   
1                                    Data Run: 4/5/2012   
2                                                         
3                                                   NaN   
649                                            00671300   
654                                            00684200   
696                                            00866600   
734                                            01018800   
764                                            01097700   
820                                            01280200   
821                                            01281100   
859                                            02207400   
868                                            02233300   
870                                            02244400   
871                                            02246000   
87

In [16]:
#Award Year Summary
print("Award Year Summary")
print("Rows with missing values in Excel:")
print(ays[ays.isna().any(axis=1)])

print("Rows with missing values in CSV:")
print(ays_csv[ays_csv.isna().any(axis=1)])


Award Year Summary
Rows with missing values in Excel:
             2009-2010 Award Year FFEL Volume by School  \
0     Award Year Cumulative Activity through Quarter...   
1                                    Data Run: 4/5/2012   
2                                                         
3                                                   NaN   
649                                            00671300   
654                                            00684200   
696                                            00866600   
734                                            01018800   
764                                            01097700   
820                                            01280200   
821                                            01281100   
859                                            02207400   
868                                            02233300   
870                                            02244400   
871                                            02246000   
87

#### Count missing values
* Both CSV and XLS files contain the same amount of missing values
* Both the Quarterly Analysis and Award Year Summary dataframes contain the same amount of missing values
* These similarities mean we must inspect the data more carefully to determine which filetype to use

In [18]:
print("Quarterly Analysis")
print("Total missing in Excel:", qa.isna().sum().sum())
print("Total missing in CSV:", qa_csv.isna().sum().sum())

print("\nAward Year Summary")
print("Total missing in Excel:", ays.isna().sum().sum())
print("Total missing in CSV:", ays_csv.isna().sum().sum())

Quarterly Analysis
Total missing in Excel: 111
Total missing in CSV: 111

Award Year Summary
Total missing in Excel: 111
Total missing in CSV: 111


# <u>Initial Exploratory Analysis: Quarterly Analysis (Q1 Dataset)</u>
#### We will perform a basic structural comparison on the original Excel file (XLS) and converted CSV file.
* .shape: Check for identical shape of the data
* .columns: View the column names of the dataframes for discrepancies
1. Check for: Same name, no extra whitespace, same order
* .dtypes: Check for identical data types of the dataframes (objects)
* .info(): A summary
* .describe(): Check for same descriptive statistics
* .head(): See the first few rows of the data
* .tail(): See the last few rows of the data
* .sample(): See random rows from the data that may not appear in a typical viewing of the data, due to the large amount of rows.

### Shape

In [21]:
print("SHAPE:")
print("Excel: ", qa.shape, "\nCSV: ", qa_csv.shape)

SHAPE:
Excel:  (3825, 25) 
CSV:  (3825, 25)


### <u>Observation:</u> Identical shape

### Columns

In [24]:
print("\nCOLUMNS:")
print("Excel: ", qa.columns, "\nCSV: ", qa_csv.columns)


COLUMNS:
Excel:  Index(['2009-2010 Award Year FFEL Volume by School', 'Unnamed: 1',
       'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6',
       'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11',
       'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16', 'Unnamed: 17', 'Unnamed: 18', 'Unnamed: 19',
       'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23',
       'Unnamed: 24'],
      dtype='object') 
CSV:  Index(['2009-2010 Award Year FFEL Volume by School', 'Unnamed: 1',
       'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6',
       'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11',
       'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16', 'Unnamed: 17', 'Unnamed: 18', 'Unnamed: 19',
       'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23',
       'Unnamed: 24'],
      dtype='object')


### <u>Observation:</u> Columns are identical in name, whitespace, and number

### DTypes

In [25]:
print("\nDTYPES:")
print("Excel: ", qa.dtypes, "\nCSV: ", qa_csv.dtypes)


DTYPES:
Excel:  2009-2010 Award Year FFEL Volume by School    object
Unnamed: 1                                    object
Unnamed: 2                                    object
Unnamed: 3                                    object
Unnamed: 4                                    object
Unnamed: 5                                    object
Unnamed: 6                                    object
Unnamed: 7                                    object
Unnamed: 8                                    object
Unnamed: 9                                    object
Unnamed: 10                                   object
Unnamed: 11                                   object
Unnamed: 12                                   object
Unnamed: 13                                   object
Unnamed: 14                                   object
Unnamed: 15                                   object
Unnamed: 16                                   object
Unnamed: 17                                   object
Unnamed: 18                  

### <u>Observation:</u> Data Types are identical in number (1-24) and type (object)

### Info

#### Excel

In [22]:
qa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3825 entries, 0 to 3824
Data columns (total 25 columns):
 #   Column                                      Non-Null Count  Dtype 
---  ------                                      --------------  ----- 
 0   2009-2010 Award Year FFEL Volume by School  3824 non-null   object
 1   Unnamed: 1                                  3821 non-null   object
 2   Unnamed: 2                                  3821 non-null   object
 3   Unnamed: 3                                  3803 non-null   object
 4   Unnamed: 4                                  3821 non-null   object
 5   Unnamed: 5                                  3822 non-null   object
 6   Unnamed: 6                                  3821 non-null   object
 7   Unnamed: 7                                  3821 non-null   object
 8   Unnamed: 8                                  3821 non-null   object
 9   Unnamed: 9                                  3821 non-null   object
 10  Unnamed: 10             

#### CSV

In [23]:
qa_csv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3825 entries, 0 to 3824
Data columns (total 25 columns):
 #   Column                                      Non-Null Count  Dtype 
---  ------                                      --------------  ----- 
 0   2009-2010 Award Year FFEL Volume by School  3824 non-null   object
 1   Unnamed: 1                                  3821 non-null   object
 2   Unnamed: 2                                  3821 non-null   object
 3   Unnamed: 3                                  3803 non-null   object
 4   Unnamed: 4                                  3821 non-null   object
 5   Unnamed: 5                                  3822 non-null   object
 6   Unnamed: 6                                  3821 non-null   object
 7   Unnamed: 7                                  3821 non-null   object
 8   Unnamed: 8                                  3821 non-null   object
 9   Unnamed: 9                                  3821 non-null   object
 10  Unnamed: 10             

### <u>Observation:</u> Both file types are identical in object type, memory usage, number of rows, number of columns, and entries

### Describe

In [28]:
#Excel
qa.describe()

2009-2010 Award Year FFEL Volume by School  \
count                                                3824   
unique                                               3824   
top     Award Year Quarterly Activity  (07/01/2009-09/...   
freq                                                    1   

                     Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4  Unnamed: 5  \
count                      3821       3821       3803       3821        3822   
unique                     3553         56       3794          8        1395   
top     ITT TECHNICAL INSTITUTE         FC  841150000    PRIVATE           1   
freq                         29        381          2       1360         160   

        Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 15  \
count         3821        3821        3821        3821  ...         3822   
unique        1399        3411        1417        3442  ...          436   
top              1           0           1           0  ...            0   
freq           158         114         143         114  ...         1352   

        Unnamed: 16  Unnamed: 17  Unnamed: 18  Unnamed: 19  Unnamed: 20  \
count          3821         3821         3821         3821         3822   
unique          457         2252          463         2281          283   
top               0            0            0            0            0   
freq           1352         1352         1352         1352         2784   

        Unnamed: 21  Unnamed: 22  Unnamed: 23  Unnamed: 24  
count          3821         3821         3821         3821  
unique          292          992          294          989  
top               0            0            0            0  
freq           2784         2784         2784         2784  

[4 rows x 25 columns]

In [29]:
#CSV
qa_csv.describe()

2009-2010 Award Year FFEL Volume by School  \
count                                                3824   
unique                                               3824   
top     Award Year Quarterly Activity  (07/01/2009-09/...   
freq                                                    1   

                     Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4 Unnamed: 5  \
count                      3821       3821       3803       3821       3822   
unique                     3553         56       3794          8       1395   
top     ITT TECHNICAL INSTITUTE         FC  841150000    PRIVATE         1    
freq                         29        381          2       1360        160   

       Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9  ... Unnamed: 15  \
count        3821       3821       3821       3821  ...        3822   
unique       1399       3411       1417       3442  ...         436   
top            1      $ -             1     $ -     ...        -      
freq          158        114        143        114  ...        1352   

       Unnamed: 16 Unnamed: 17 Unnamed: 18 Unnamed: 19 Unnamed: 20  \
count         3821        3821        3821        3821        3822   
unique         457        2252         463        2281         283   
top           -         $ -              0      $ -           -      
freq          1352        1352        1352        1352        2784   

       Unnamed: 21 Unnamed: 22 Unnamed: 23 Unnamed: 24  
count         3821        3821        3821        3821  
unique         292         992         294         989  
top           -         $ -              0      $ -     
freq          2784        2784        2784        2784  

[4 rows x 25 columns]

### <u>Discrepancy Discovered:</u> 
* Columns names have been changed in each dataframe. "School" is now "Unnamed: 1"
* ITT row contains values of 0 in XLS file, and $- in CSV file.
### Suggestion: Use the XLS sheet file (qa), which lies inside of a dictionary for the entire dataframe.
1. Rename columns manually by comparing number of column to column name on XLS sheet.
2. Drop all rows containing a value of '0' so there are no empty values where there should be data.

### View the ITT row

In [32]:
#From Excel DataFrame
itt_excel = qa[qa["Unnamed: 1"] == "ITT TECHNICAL INSTITUTE"]

#From CSV DataFrame
itt_csv = qa_csv[qa_csv["Unnamed: 1"] == "ITT TECHNICAL INSTITUTE"]

### Print both rows

In [34]:
print("Excel version of ITT row:")
print(itt_excel)

Excel version of ITT row:
     2009-2010 Award Year FFEL Volume by School               Unnamed: 1  \
111                                    02065200  ITT TECHNICAL INSTITUTE   
127                                    02361100  ITT TECHNICAL INSTITUTE   
366                                    02120900  ITT TECHNICAL INSTITUTE   
386                                    02291500  ITT TECHNICAL INSTITUTE   
387                                    02291600  ITT TECHNICAL INSTITUTE   
392                                    02321800  ITT TECHNICAL INSTITUTE   
393                                    02321900  ITT TECHNICAL INSTITUTE   
439                                    03070400  ITT TECHNICAL INSTITUTE   
443                                    03087400  ITT TECHNICAL INSTITUTE   
550                                    02321700  ITT TECHNICAL INSTITUTE   
1087                                   02286500  ITT TECHNICAL INSTITUTE   
1112                                   03087600  ITT TECHNICAL

In [35]:
print("\nCSV version of ITT row:")
print(itt_csv)


CSV version of ITT row:
     2009-2010 Award Year FFEL Volume by School               Unnamed: 1  \
111                                    02065200  ITT TECHNICAL INSTITUTE   
127                                    02361100  ITT TECHNICAL INSTITUTE   
366                                    02120900  ITT TECHNICAL INSTITUTE   
386                                    02291500  ITT TECHNICAL INSTITUTE   
387                                    02291600  ITT TECHNICAL INSTITUTE   
392                                    02321800  ITT TECHNICAL INSTITUTE   
393                                    02321900  ITT TECHNICAL INSTITUTE   
439                                    03070400  ITT TECHNICAL INSTITUTE   
443                                    03087400  ITT TECHNICAL INSTITUTE   
550                                    02321700  ITT TECHNICAL INSTITUTE   
1087                                   02286500  ITT TECHNICAL INSTITUTE   
1112                                   03087600  ITT TECHNICAL 

### <u>Observation:</u> The school name appears several times in the file. 
* Suggestion: For each school name, add all values together and merge into a single row.

### Heads
#### Excel

In [4]:
qa.head()

2009-2010 Award Year FFEL Volume by School Unnamed: 1 Unnamed: 2  \
0  Award Year Quarterly Activity  (07/01/2009-09/...        NaN        NaN   
1                                 Data Run: 4/5/2012        NaN        NaN   
2                                                           NaN        NaN   
3                                                NaN        NaN        NaN   
4                                             OPE ID     School      State   

  Unnamed: 3   Unnamed: 4       Unnamed: 5             Unnamed: 6  \
0        NaN          NaN              NaN                    NaN   
1        NaN          NaN              NaN                    NaN   
2        NaN          NaN              NaN                    NaN   
3        NaN          NaN  FFEL SUBSIDIZED                    NaN   
4   Zip Code  School Type       Recipients  # of Loans Originated   

              Unnamed: 7          Unnamed: 8          Unnamed: 9  ...  \
0                    NaN                 NaN                 NaN  ...   
1                    NaN                 NaN                 NaN  ...   
2                    NaN                 NaN                 NaN  ...   
3                    NaN                 NaN                 NaN  ...   
4  $ of Loans Originated  # of Disbursements  $ of Disbursements  ...   

        Unnamed: 15            Unnamed: 16            Unnamed: 17  \
0               NaN                    NaN                    NaN   
1               NaN                    NaN                    NaN   
2               NaN                    NaN                    NaN   
3  FFEL PARENT PLUS                    NaN                    NaN   
4        Recipients  # of Loans Originated  $ of Loans Originated   

          Unnamed: 18         Unnamed: 19     Unnamed: 20  \
0                 NaN                 NaN             NaN   
1                 NaN                 NaN             NaN   
2                 NaN                 NaN             NaN   
3                 NaN                 NaN  FFEL GRAD PLUS   
4  # of Disbursements  $ of Disbursements      Recipients   

             Unnamed: 21            Unnamed: 22         Unnamed: 23  \
0                    NaN                    NaN                 NaN   
1                    NaN                    NaN                 NaN   
2                    NaN                    NaN                 NaN   
3                    NaN                    NaN                 NaN   
4  # of Loans Originated  $ of Loans Originated  # of Disbursements   

          Unnamed: 24  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4  $ of Disbursements  

[5 rows x 25 columns]

### CSV

In [37]:
qa_csv.head()

2009-2010 Award Year FFEL Volume by School Unnamed: 1 Unnamed: 2  \
0  Award Year Quarterly Activity  (07/01/2009-09/...        NaN        NaN   
1                                 Data Run: 4/5/2012        NaN        NaN   
2                                                           NaN        NaN   
3                                                NaN        NaN        NaN   
4                                             OPE ID     School      State   

  Unnamed: 3   Unnamed: 4         Unnamed: 5             Unnamed: 6  \
0        NaN          NaN                NaN                    NaN   
1        NaN          NaN                NaN                    NaN   
2        NaN          NaN                NaN                    NaN   
3        NaN          NaN   FFEL SUBSIDIZED                     NaN   
4   Zip Code  School Type         Recipients  # of Loans Originated   

              Unnamed: 7          Unnamed: 8          Unnamed: 9  ...  \
0                    NaN                 NaN                 NaN  ...   
1                    NaN                 NaN                 NaN  ...   
2                    NaN                 NaN                 NaN  ...   
3                    NaN                 NaN                 NaN  ...   
4  $ of Loans Originated  # of Disbursements  $ of Disbursements  ...   

          Unnamed: 15            Unnamed: 16            Unnamed: 17  \
0                 NaN                    NaN                    NaN   
1                 NaN                    NaN                    NaN   
2                 NaN                    NaN                    NaN   
3   FFEL PARENT PLUS                     NaN                    NaN   
4          Recipients  # of Loans Originated  $ of Loans Originated   

          Unnamed: 18         Unnamed: 19     Unnamed: 20  \
0                 NaN                 NaN             NaN   
1                 NaN                 NaN             NaN   
2                 NaN                 NaN             NaN   
3                 NaN                 NaN  FFEL GRAD PLUS   
4  # of Disbursements  $ of Disbursements      Recipients   

             Unnamed: 21            Unnamed: 22         Unnamed: 23  \
0                    NaN                    NaN                 NaN   
1                    NaN                    NaN                 NaN   
2                    NaN                    NaN                 NaN   
3                    NaN                    NaN                 NaN   
4  # of Loans Originated  $ of Loans Originated  # of Disbursements   

          Unnamed: 24  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4  $ of Disbursements  

[5 rows x 25 columns]

### <u>Observation:</u> Both dataframes look identical, but contain missing values.
* The first few rows of each dataframe are missing.
* The categories of SUBSIDIZED and UNSUBSIDIZED lie within rows that are missing.
* The columns, which should represent things like "Zip Code, School Type" are "Unnamed: 1", "Unnamed: 2"...and so on.
* The proper column names are in row 4 (index 0, so it is index[4]). 

### Sample random rows
* Sample 5 random rows from each dataset
* Sample rows by index for direct comparison

#### Sample 5 random rows

In [39]:
print("\n5 RANDOM SAMPLES:")
print("Excel: ")
qa.sample(5)


5 RANDOM SAMPLES:
Excel: 


2009-2010 Award Year FFEL Volume by School  \
2178                                   00296400   
563                                    00137900   
1862                                   00227800   
1735                                   00748800   
2659                                   00310000   

                          Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4  \
2178  SOUTHEASTERN COMMUNITY COLLEGE         NC  284725422       PUBLIC   
563              CONNECTICUT COLLEGE         CT  063204196      PRIVATE   
1862       LANSING COMMUNITY COLLEGE         MI  489331293       PUBLIC   
1735         KAPLAN CAREER INSTITUTE         MA  021291644  PROPRIETARY   
2659                 OHIO UNIVERSITY         OH  457012979       PUBLIC   

     Unnamed: 5 Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9  ... Unnamed: 15  \
2178        236        237     675452        237     439470  ...           0   
563         154        154     735182        154     367341  ...          33   
1862          1          1       1522          1       1750  ...           0   
1735         74         74     240350         75     104458  ...           3   
2659        209        209    1768415        209     591493  ...           0   

     Unnamed: 16 Unnamed: 17 Unnamed: 18 Unnamed: 19 Unnamed: 20 Unnamed: 21  \
2178           0           0           0           0           0           0   
563           34      767830          34      383919           0           0   
1862           0           0           0           0           0           0   
1735           3        9410           3        4821           0           0   
2659           0           0           0           0          43          47   

     Unnamed: 22 Unnamed: 23 Unnamed: 24  
2178           0           0           0  
563            0           0           0  
1862           0           0           0  
1735           0           0           0  
2659      472208          48      164999  

[5 rows x 25 columns]

In [41]:
print("\n5 RANDOM SAMPLES:")
print("CSV: ")
qa_csv.sample(5)


5 RANDOM SAMPLES:
CSV: 


2009-2010 Award Year FFEL Volume by School  \
416                                    02584200   
2993                                   00531000   
1811                                   03296300   
1681                                   00212600   
452                                    03131300   

                               Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4  \
416                     NEWBRIDGE COLLEGE         CA  927058605  PROPRIETARY   
2993  PITTSBURGH INSTITUTE OF AERONAUTICS         PA  151222674      PRIVATE   
1811          BALTIMORE SCHOOL OF MASSAGE         MD  210902261  PROPRIETARY   
1681             BERKLEE COLLEGE OF MUSIC         MA  022153693      PRIVATE   
452              FIVE BRANCHES UNIVERSITY         CA  950624669      PRIVATE   

     Unnamed: 5 Unnamed: 6        Unnamed: 7 Unnamed: 8      Unnamed: 9  ...  \
416        115        115      $ 419,821.00         122   $ 218,028.00   ...   
2993        63         63      $ 233,820.00          63    $ 91,557.00   ...   
1811       125        125      $ 366,676.00         127   $ 137,109.00   ...   
1681       -          -               $ -             0          $ -     ...   
452        138        138    $ 1,120,675.00         139   $ 625,007.00   ...   

     Unnamed: 15 Unnamed: 16     Unnamed: 17 Unnamed: 18     Unnamed: 19  \
416           5           5     $ 21,645.00            5    $ 12,113.00    
2993          9          10    $ 126,786.00           10    $ 43,431.00    
1811          6           6     $ 40,609.00            6    $ 13,543.00    
1681          9           9    $ 186,935.00            9   $ 105,918.00    
452         -           -             $ -              0          $ -      

     Unnamed: 20 Unnamed: 21       Unnamed: 22 Unnamed: 23     Unnamed: 24  
416         -           -               $ -              0          $ -     
2993        -           -               $ -              0          $ -     
1811        -           -               $ -              0          $ -     
1681        -           -               $ -              0          $ -     
452          58          59    $ 1,217,611.00           68   $ 580,365.00   

[5 rows x 25 columns]

### Sample rows by index
### Excel
* Define the row numbers as a list
* Use "loc" to view the specified rows

In [43]:
indices = [10, 42, 77]
print("Excel sample:")
qa.loc[indices]

Excel sample:


2009-2010 Award Year FFEL Volume by School                     Unnamed: 1  \
10                                   02576900                CHARTER COLLEGE   
42                                   02270400     SOUTHEASTERN BIBLE COLLEGE   
77                                   02052200  BLACK RIVER TECHNICAL COLLEGE   

   Unnamed: 2 Unnamed: 3   Unnamed: 4 Unnamed: 5 Unnamed: 6 Unnamed: 7  \
10         AK  995084103  PROPRIETARY        192        192     516838   
42         AL  352442083      PRIVATE        123        124     495343   
77         AR  724550468       PUBLIC        648        662    1965082   

   Unnamed: 8 Unnamed: 9  ... Unnamed: 15 Unnamed: 16 Unnamed: 17 Unnamed: 18  \
10        193     255514  ...           4           4       20866           5   
42        127     259886  ...           3           3       34500           3   
77        669     998305  ...           0           0           0           0   

   Unnamed: 19 Unnamed: 20 Unnamed: 21 Unnamed: 22 Unnamed: 23 Unnamed: 24  
10       12783           0           0           0           0           0  
42       17250           0           0           0           0           0  
77           0           0           0           0           0           0  

[3 rows x 25 columns]

### CSV

In [44]:
print("CSV sample:")
qa_csv.loc[indices]

CSV sample:


2009-2010 Award Year FFEL Volume by School                     Unnamed: 1  \
10                                   02576900                CHARTER COLLEGE   
42                                   02270400     SOUTHEASTERN BIBLE COLLEGE   
77                                   02052200  BLACK RIVER TECHNICAL COLLEGE   

   Unnamed: 2 Unnamed: 3   Unnamed: 4 Unnamed: 5 Unnamed: 6        Unnamed: 7  \
10         AK  995084103  PROPRIETARY       192        192      $ 516,838.00    
42         AL  352442083      PRIVATE       123        124      $ 495,343.00    
77         AR  724550468       PUBLIC       648        662    $ 1,965,082.00    

   Unnamed: 8      Unnamed: 9  ... Unnamed: 15 Unnamed: 16    Unnamed: 17  \
10        193   $ 255,514.00   ...          4           4    $ 20,866.00    
42        127   $ 259,886.00   ...          3           3    $ 34,500.00    
77        669   $ 998,305.00   ...        -           -            $ -      

   Unnamed: 18    Unnamed: 19 Unnamed: 20 Unnamed: 21 Unnamed: 22 Unnamed: 23  \
10           5   $ 12,783.00         -           -         $ -              0   
42           3   $ 17,250.00         -           -         $ -              0   
77           0         $ -           -           -         $ -              0   

   Unnamed: 24  
10      $ -     
42      $ -     
77      $ -     

[3 rows x 25 columns]

### <u>Observation:</u> The Excel file contains....
* Integers, not decimals
* No dollar signs
* 0 in place of missing values, instead of "-"
### The CSV file contains...
* Decimals, with dollar signs
* "-", "$-", and 0 in place of missing values

### <u> Suggestion:</u> Use original XLS file.
#### Reasoning:
* Cleaner and more consistent structure
* No extra formatting artifacts like "$" or "-" (difficult to index and impossible to calculate, difficult to drop)
* Easier to process
* Already structured as a dictionary of sheets (qa and ays are the keys in the dictionary for each sheet)
* Missing values appear to be consistently marked as 0 (easier to or exclude from averages)

### For the reasons above, we will be using the XLS files (qa and ays) exclusively from now on and disregarding the csv files.
### Tail: View the tail end of the structure (the last few rows)

In [3]:
qa.tail()

2009-2010 Award Year FFEL Volume by School  \
3820                                   00393200   
3821                                   00393300   
3822                                   00728900   
3823                                   00915700   
3824                                   00925900   

                             Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4  \
3820              UNIVERSITY OF WYOMING         WY  820713663       PUBLIC   
3821  WESTERN WYOMING COMMUNITY COLLEGE         WY  829010428       PUBLIC   
3822            CENTRAL WYOMING COLLEGE         WY  825012215       PUBLIC   
3823                            WYOTECH         WY  820729519  PROPRIETARY   
3824   LARAMIE COUNTY COMMUNITY COLLEGE         WY  820073299       PUBLIC   

     Unnamed: 5 Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9  ... Unnamed: 15  \
3820       2699       2735   12341711       2738    6819888  ...         195   
3821        214        214     731883        215     359742  ...           2   
3822        149        154     414959        154     212783  ...           7   
3823       1049       1099    2554580       1220    1498483  ...         478   
3824        281        291     885794        291     447794  ...           8   

     Unnamed: 16 Unnamed: 17 Unnamed: 18 Unnamed: 19 Unnamed: 20 Unnamed: 21  \
3820         207     1830536         209      950682          58          65   
3821           2       13300           2        6650           0           0   
3822           7       34978           7       17492           0           0   
3823         490     7567391         522     4033026           0           0   
3824           8       41636           8       20818           0           0   

     Unnamed: 22 Unnamed: 23 Unnamed: 24  
3820      740877          66      416236  
3821           0           0           0  
3822           0           0           0  
3823           0           0           0  
3824           0           0           0  

[5 rows x 25 columns]

### <u>Observation:</u> The XLS file will be used with caution.
* Columns must be renamed from (Unnamed: 1) to appropriate name matching the original XLS file
* Values contain integers only, no decimals. Operations such as division must be performed carefully.
* Missing values contain "0" and may be dropped or excluded from analysis

# <u>Initial Exploratory Analysis: Award Year Summary (Q1 Dataset)</u>
### We will perform a basic structural comparison on the XLS file.
* .shape: Determine the shape of the data
* .columns: View the column names
* .dtypes: Determine the datatypes of the data
* .info(): A summary
* .describe(): Check the descriptive statistics
* .head(): See the first few rows of the data
* .tail(): See the last few rows of the data
* .sample(): See random rows from the data that may not appear in a typical viewing of the data, due to the large amount of rows.
1. Sample 5 random rows
2. Sample a specific index of rows

### Shape
* Observation: The shape is identical to the Quarterly Analysis sheet

In [47]:
ays.shape

(3825, 25)

### Columns
* Observation: The column names are identical to the Quarterly Analysis sheet. They do not match the column names in the original Excel file, such as "School" and "State".

In [50]:
ays.columns

Index(['2009-2010 Award Year FFEL Volume by School', 'Unnamed: 1',
       'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6',
       'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11',
       'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14', 'Unnamed: 15',
       'Unnamed: 16', 'Unnamed: 17', 'Unnamed: 18', 'Unnamed: 19',
       'Unnamed: 20', 'Unnamed: 21', 'Unnamed: 22', 'Unnamed: 23',
       'Unnamed: 24'],
      dtype='object')

### Data types
* Observation: Same number (24) and type of data (objects) as the QA sheet.

In [52]:
ays.dtypes

2009-2010 Award Year FFEL Volume by School    object
Unnamed: 1                                    object
Unnamed: 2                                    object
Unnamed: 3                                    object
Unnamed: 4                                    object
Unnamed: 5                                    object
Unnamed: 6                                    object
Unnamed: 7                                    object
Unnamed: 8                                    object
Unnamed: 9                                    object
Unnamed: 10                                   object
Unnamed: 11                                   object
Unnamed: 12                                   object
Unnamed: 13                                   object
Unnamed: 14                                   object
Unnamed: 15                                   object
Unnamed: 16                                   object
Unnamed: 17                                   object
Unnamed: 18                                   

### Info
### Observations: Info appears identical to QA Sheet...
* Column name and number
* D type
* Number of non-null objects
* Memory usage
* Number of entries

In [53]:
ays.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3825 entries, 0 to 3824
Data columns (total 25 columns):
 #   Column                                      Non-Null Count  Dtype 
---  ------                                      --------------  ----- 
 0   2009-2010 Award Year FFEL Volume by School  3824 non-null   object
 1   Unnamed: 1                                  3821 non-null   object
 2   Unnamed: 2                                  3821 non-null   object
 3   Unnamed: 3                                  3803 non-null   object
 4   Unnamed: 4                                  3821 non-null   object
 5   Unnamed: 5                                  3822 non-null   object
 6   Unnamed: 6                                  3821 non-null   object
 7   Unnamed: 7                                  3821 non-null   object
 8   Unnamed: 8                                  3821 non-null   object
 9   Unnamed: 9                                  3821 non-null   object
 10  Unnamed: 10             

### Descriptive Statistics

In [54]:
ays.describe()

2009-2010 Award Year FFEL Volume by School  \
count                                                3824   
unique                                               3824   
top     Award Year Cumulative Activity through Quarter...   
freq                                                    1   

                     Unnamed: 1 Unnamed: 2 Unnamed: 3 Unnamed: 4  Unnamed: 5  \
count                      3821       3821       3803       3821        3822   
unique                     3553         56       3794          8        1395   
top     ITT TECHNICAL INSTITUTE         FC  841150000    PRIVATE           1   
freq                         29        381          2       1360         160   

        Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 15  \
count         3821        3821        3821        3821  ...         3822   
unique        1399        3411        1417        3442  ...          436   
top              1           0           1           0  ...            0   
freq           158         114         143         114  ...         1352   

        Unnamed: 16  Unnamed: 17  Unnamed: 18  Unnamed: 19  Unnamed: 20  \
count          3821         3821         3821         3821         3822   
unique          457         2252          463         2281          283   
top               0            0            0            0            0   
freq           1352         1352         1352         1352         2784   

        Unnamed: 21  Unnamed: 22  Unnamed: 23  Unnamed: 24  
count          3821         3821         3821         3821  
unique          292          992          294          989  
top               0            0            0            0  
freq           2784         2784         2784         2784  

[4 rows x 25 columns]

### Observations:
* ITT appears 29 times throughout the data.
* FC is listed as a state. This likely means "Foreign Country", as in the applicant of the loan is from a foreign country.
* The reason the data is missing for FC is likely because it is difficult to get data on foreign students.

### Head

In [55]:
ays.head()

2009-2010 Award Year FFEL Volume by School Unnamed: 1 Unnamed: 2  \
0  Award Year Cumulative Activity through Quarter...        NaN        NaN   
1                                 Data Run: 4/5/2012        NaN        NaN   
2                                                           NaN        NaN   
3                                                NaN        NaN        NaN   
4                                             OPE ID     School      State   

  Unnamed: 3   Unnamed: 4       Unnamed: 5             Unnamed: 6  \
0        NaN          NaN              NaN                    NaN   
1        NaN          NaN              NaN                    NaN   
2        NaN          NaN              NaN                    NaN   
3        NaN          NaN  FFEL SUBSIDIZED                    NaN   
4   Zip Code  School Type       Recipients  # of Loans Originated   

              Unnamed: 7          Unnamed: 8          Unnamed: 9  ...  \
0                    NaN                 NaN                 NaN  ...   
1                    NaN                 NaN                 NaN  ...   
2                    NaN                 NaN                 NaN  ...   
3                    NaN                 NaN                 NaN  ...   
4  $ of Loans Originated  # of Disbursements  $ of Disbursements  ...   

        Unnamed: 15            Unnamed: 16            Unnamed: 17  \
0               NaN                    NaN                    NaN   
1               NaN                    NaN                    NaN   
2               NaN                    NaN                    NaN   
3  FFEL PARENT PLUS                    NaN                    NaN   
4        Recipients  # of Loans Originated  $ of Loans Originated   

          Unnamed: 18         Unnamed: 19     Unnamed: 20  \
0                 NaN                 NaN             NaN   
1                 NaN                 NaN             NaN   
2                 NaN                 NaN             NaN   
3                 NaN                 NaN  FFEL GRAD PLUS   
4  # of Disbursements  $ of Disbursements      Recipients   

             Unnamed: 21            Unnamed: 22         Unnamed: 23  \
0                    NaN                    NaN                 NaN   
1                    NaN                    NaN                 NaN   
2                    NaN                    NaN                 NaN   
3                    NaN                    NaN                 NaN   
4  # of Loans Originated  $ of Loans Originated  # of Disbursements   

          Unnamed: 24  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4  $ of Disbursements  

[5 rows x 25 columns]

### Tail

In [56]:
ays.tail()

2009-2010 Award Year FFEL Volume by School  \
3820                                   00393200   
3821                                   00393300   
3822                                   00728900   
3823                                   00915700   
3824                                   00925900   

                             Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4  \
3820              UNIVERSITY OF WYOMING         WY  820713663       PUBLIC   
3821  WESTERN WYOMING COMMUNITY COLLEGE         WY  829010428       PUBLIC   
3822            CENTRAL WYOMING COLLEGE         WY  825012215       PUBLIC   
3823                            WYOTECH         WY  820729519  PROPRIETARY   
3824   LARAMIE COUNTY COMMUNITY COLLEGE         WY  820073299       PUBLIC   

     Unnamed: 5 Unnamed: 6 Unnamed: 7 Unnamed: 8 Unnamed: 9  ... Unnamed: 15  \
3820       2699       2735   12341711       2738    6819888  ...         195   
3821        214        214     731883        215     359742  ...           2   
3822        149        154     414959        154     212783  ...           7   
3823       1049       1099    2554580       1220    1498483  ...         478   
3824        281        291     885794        291     447794  ...           8   

     Unnamed: 16 Unnamed: 17 Unnamed: 18 Unnamed: 19 Unnamed: 20 Unnamed: 21  \
3820         207     1830536         209      950682          58          65   
3821           2       13300           2        6650           0           0   
3822           7       34978           7       17492           0           0   
3823         490     7567391         522     4033026           0           0   
3824           8       41636           8       20818           0           0   

     Unnamed: 22 Unnamed: 23 Unnamed: 24  
3820      740877          66      416236  
3821           0           0           0  
3822           0           0           0  
3823           0           0           0  
3824           0           0           0  

[5 rows x 25 columns]

### 5 Random Samples

### Sampled index
* We will use the same index we defined when viewing a sampled index of the Quarterly Analysis data sheet

In [49]:
ays.loc[indices]

2009-2010 Award Year FFEL Volume by School                     Unnamed: 1  \
10                                   02576900                CHARTER COLLEGE   
42                                   02270400     SOUTHEASTERN BIBLE COLLEGE   
77                                   02052200  BLACK RIVER TECHNICAL COLLEGE   

   Unnamed: 2 Unnamed: 3   Unnamed: 4 Unnamed: 5 Unnamed: 6 Unnamed: 7  \
10         AK  995084103  PROPRIETARY        192        192     516838   
42         AL  352442083      PRIVATE        123        124     495343   
77         AR  724550468       PUBLIC        648        662    1965082   

   Unnamed: 8 Unnamed: 9  ... Unnamed: 15 Unnamed: 16 Unnamed: 17 Unnamed: 18  \
10        193     255514  ...           4           4       20866           5   
42        127     259886  ...           3           3       34500           3   
77        669     998305  ...           0           0           0           0   

   Unnamed: 19 Unnamed: 20 Unnamed: 21 Unnamed: 22 Unnamed: 23 Unnamed: 24  
10       12783           0           0           0           0           0  
42       17250           0           0           0           0           0  
77           0           0           0           0           0           0  

[3 rows x 25 columns]

### View the ITT Row

In [57]:
ays_itt = ays[ays["Unnamed: 1"] == "ITT TECHNICAL INSTITUTE"]

In [58]:
ays_itt

2009-2010 Award Year FFEL Volume by School               Unnamed: 1  \
111                                    02065200  ITT TECHNICAL INSTITUTE   
127                                    02361100  ITT TECHNICAL INSTITUTE   
366                                    02120900  ITT TECHNICAL INSTITUTE   
386                                    02291500  ITT TECHNICAL INSTITUTE   
387                                    02291600  ITT TECHNICAL INSTITUTE   
392                                    02321800  ITT TECHNICAL INSTITUTE   
393                                    02321900  ITT TECHNICAL INSTITUTE   
439                                    03070400  ITT TECHNICAL INSTITUTE   
443                                    03087400  ITT TECHNICAL INSTITUTE   
550                                    02321700  ITT TECHNICAL INSTITUTE   
1087                                   02286500  ITT TECHNICAL INSTITUTE   
1112                                   03087600  ITT TECHNICAL INSTITUTE   
1306                                   00455300  ITT TECHNICAL INSTITUTE   
1489                                   00732700  ITT TECHNICAL INSTITUTE   
1490                                   00732900  ITT TECHNICAL INSTITUTE   
1493                                   00832900  ITT TECHNICAL INSTITUTE   
1891                                   01062700  ITT TECHNICAL INSTITUTE   
2046                                   00755700  ITT TECHNICAL INSTITUTE   
2695                                   00908800  ITT TECHNICAL INSTITUTE   
2700                                   00983700  ITT TECHNICAL INSTITUTE   
2854                                   01185200  ITT TECHNICAL INSTITUTE   
3294                                   02359800  ITT TECHNICAL INSTITUTE   
3301                                   03073400  ITT TECHNICAL INSTITUTE   
3463                                   02328600  ITT TECHNICAL INSTITUTE   
3495                                   03071400  ITT TECHNICAL INSTITUTE   
3541                                   02361000  ITT TECHNICAL INSTITUTE   
3695                                   00844300  ITT TECHNICAL INSTITUTE   
3717                                   03071800  ITT TECHNICAL INSTITUTE   
3783                                   03087500  ITT TECHNICAL INSTITUTE   

     Unnamed: 2 Unnamed: 3   Unnamed: 4 Unnamed: 5 Unnamed: 6 Unnamed: 7  \
111          AZ  852829272  PROPRIETARY        425        432    1741803   
127          AZ  857045829  PROPRIETARY        338        343    1332796   
366          CA  956706047  PROPRIETARY        555        556    2025625   
386          CA  917732933  PROPRIETARY        259        271    1023487   
387          CA  921232662  PROPRIETARY        556        558    2017752   
392          CA  913423664  PROPRIETARY        511        514    1932110   
393          CA  928015454  PROPRIETARY        385        386    1417241   
439          CA  924083519  PROPRIETARY        563        565    2082138   
443          CA  905021356  PROPRIETARY        331        334    1351752   
550          CO  802295338  PROPRIETARY        391        391    1452092   
1087         FL  336347515  PROPRIETARY        495        500    2015140   
1112         FL  327461607  PROPRIETARY       1109       1119    4041940   
1306         ID  837131529  PROPRIETARY        883        887    3267305   
1489         IN  476307430  PROPRIETARY        742        751    2917702   
1490         IN  462681119  PROPRIETARY       2475       2483    9202399   
1493         IN  468252393  PROPRIETARY       1568       1580    5830663   
1891         MI  495190000  PROPRIETARY       1007       1015    3680832   
2046         MO  630451122  PROPRIETARY        726        732    2845183   
2695         OH  454143456  PROPRIETARY        352        395    1292585   
2700         OH  445094098  PROPRIETARY        504        509    1800475   
2854         OR  972206819  PROPRIETARY        255        255    1005205   
3294         TN  372143717  PROPRIETARY        410  

### <u>Observation:</u> ITT is listed under several different states and appears multiple times in the AYS data.
* Suggestion: Add up all value data (recipient number, loan disbursement amount, etc.) per school and merge rows.

# <u>Initial Exploratory Analysis: Federal School List</u>
## Step 1: Load and view raw dataframe

In [3]:
fsl = pd.read_excel("1617fedschoolcodelist.xls")

C:\Users\pmost\anaconda3\envs\first\Lib\site-packages\openpyxl\worksheet\_reader.py:329: UserWarning: Unknown extension is not supported and will be removed
  warn(msg)


In [4]:
fsl

SchoolCode                                SchoolName  \
0        B04724           WIDENER UNIV SCHOOL OF LAW - DE   
1        B06171       CENTER FOR ADVANCED STUDIES OF PUER   
2        B06511          PENTECOSTAL THEOLOGICAL SEMINARY   
3        B07022     THE CHICAGO SCHOOL OF PROF PSYCHOLOGY   
4        B07624      NATIONAL COLLEGE OF NATURAL MEDICINE   
...         ...                                       ...   
6975     042517         HOPE COLLEGE OF ARTS AND SCIENCES   
6976     E40419  BEAUTY INSTITUTE SCHWARZKOPF PROFESSIONA   
6977     042205  BUTTE COUNTY REGIONAL OCCUPATIONAL PROGR   
6978     G42404  UNIVERSIDAD ANA G. MENDEZ - CAMPUS VIRTU   
6979     042467          LEARNING BRIDGE CAREER INSTITUTE   

                            Address           City StateCode  ZipCode  \
0     4601 CONCORD PIKE/PO BOX 7474     WILMINGTON        DE    19803   
1                        BOX S-4467       SAN JUAN        PR      902   
2                       PO BOX 3330      CLEVELAND        TN    37320   
3            325 NORTH WELLS STREET        CHICAGO        IL    60610   
4                     049 SW PORTER       PORTLAND        OR    97201   
...                             ...            ...       ...      ...   
6975     1200 SOUTH WEST 3RD STREET  POMPANO BEACH        FL    33069   
6976           1411 RAILROAD AVENUE     BELLINGHAM        WA    98225   
6977          2491 CARMICHAEL DRIVE          CHICO        CA    95928   
6978      1552 AVENUE PONCE DE LEON       SAN JUAN        PR      926   
6979    1340 WEST TUNNELL BOULEVARD          HOUMA        LA    70360   

     Province Country PostalCode  
0         NaN     NaN        NaN  
1         NaN     NaN        NaN  
2         NaN     NaN        NaN  
3         NaN     NaN        NaN  
4         NaN     NaN        NaN  
...       ...     ...        ...  
6975      NaN     NaN        NaN  
6976      NaN     NaN        NaN  
6977      NaN     NaN        NaN  
6978      NaN     NaN        NaN  
6979      NaN     NaN        NaN  

[6980 rows x 9 columns]

### Observations: This file....
* Contains school codes that have a mix of letters and numbers. The school codes are not present in any other data sheets.
* Contains schools from Puerto Rico, which is treated as a US State
* Contains schools from other countries, where they contain values for the columns "Province", "Country", and "PostalCode".
### Suggestions:
* Delete or disregard any rows containing school codes that are not in the QA or AYS (School codes containing letters, like B04...)
* Delete rows containing schools from PR, as we are only looking at info on US States, and PR is a territory.
* Any row that contains a non-missing value for "Country" should be deleted, since these are foreign schools. We will only investigate schools in US States.
* The "province", "country", and "postalcode" columns can be deleted entirely since only foreign schools show values in these columns.

### Columns

### Observation: Column names match original Excel sheet, but whitespace is removed and title case is observed.

In [5]:
fsl.columns

Index(['SchoolCode', 'SchoolName', 'Address', 'City', 'StateCode', 'ZipCode',
       'Province', 'Country', 'PostalCode'],
      dtype='object')

### Inspect ITT Rows

In [6]:
fsl_itt = fsl[fsl["SchoolName"] == "ITT TECHNICAL INSTITUTE"]


In [7]:
fsl_itt

SchoolCode               SchoolName                          Address  \
148      E00786  ITT TECHNICAL INSTITUTE            4520 SOUTH UNIVERSITY   
171      E00882  ITT TECHNICAL INSTITUTE                  168 GIBSON ROAD   
187      E00922  ITT TECHNICAL INSTITUTE                 13 AIRLINE DRIVE   
224      E01017  ITT TECHNICAL INSTITUTE                   760 MOORE ROAD   
508      E01966  ITT TECHNICAL INSTITUTE              17390 DUGDALE DRIVE   
509      E01967  ITT TECHNICAL INSTITUTE              8488 GEORGIA STREET   
3245     004553  ITT TECHNICAL INSTITUTE        12302 WEST EXPLORER DRIVE   
3616     007327  ITT TECHNICAL INSTITUTE                 10999 STAHL ROAD   
3617     007329  ITT TECHNICAL INSTITUTE                9511 ANGOLA COURT   
3662     007557  ITT TECHNICAL INSTITUTE       3640 CORPORATE TRAIL DRIVE   
3772     008329  ITT TECHNICAL INSTITUTE       2810 DUPONT COMMERCE COURT   
3852     009088  ITT TECHNICAL INSTITUTE             3325 STOP EIGHT ROAD   
3960     009837  ITT TECHNICAL INSTITUTE         1030 NORTH MERIDIAN ROAD   
4086     010627  ITT TECHNICAL INSTITUTE              1980 METRO COURT SW   
4205     011852  ITT TECHNICAL INSTITUTE  9500 NORTHEAST CASCADES PARKWAY   
4473     013928  ITT TECHNICAL INSTITUTE         5005 SOUTH WENDLER DRIVE   
4631     014685  ITT TECHNICAL INSTITUTE          10863 GOLD CENTER DRIVE   
5001     016204  ITT TECHNICAL INSTITUTE            4809 MEMORIAL HIGHWAY   
5002     016205  ITT TECHNICAL INSTITUTE         9680 GRANITE RIDGE DRIVE   
5003     016206  ITT TECHNICAL INSTITUTE          650 WEST CIENEGA AVENUE   
5052     016429  ITT TECHNICAL INSTITUTE           12669 ENCINITAS AVENUE   
5053     016430  ITT TECHNICAL INSTITUTE     4000 WEST METROPOLITAN DRIVE   
5070     016496  ITT TECHNICAL INSTITUTE             551 RYAN PLAZA DRIVE   
5071     016497  ITT TECHNICAL INSTITUTE               2950 SOUTH GESSNER   
5073     016500  ITT TECHNICAL INSTITUTE                500 E 84TH AVENUE   
5155     016767  ITT TECHNICAL INSTITUTE               2845 ELM HILL PIKE   
5157     016774  ITT TECHNICAL INSTITUTE             920 WEST LEVOY DRIVE   
5159     016776  ITT TECHNICAL INSTITUTE             1455 WEST RIVER ROAD   
5566     030704  ITT TECHNICAL INSTITUTE          670 EAST CARNEGIE DRIVE   
5569     030714  ITT TECHNICAL INSTITUTE           5700 NORTHWEST PARKWAY   
5571     030718  ITT TECHNICAL INSTITUTE        13518 EAST INDIANA AVENUE   
5578     030734  ITT TECHNICAL INSTITUTE           10208 TECHNOLOGY DRIVE   
5610     030874  ITT TECHNICAL INSTITUTE       20050 SOUTH VERMONT AVENUE   
5611     030875  ITT TECHNICAL INSTITUTE          6300 WEST LAYTON AVENUE   
5612     030876  ITT TECHNICAL INSTITUTE       1400 INTERNATIONAL PARKWAY   

                 City StateCode  ZipCode Province Country PostalCode  
148       LITTLE ROCK        AR    72204      NaN     NaN        NaN  
171         HENDERSON        NV    89014      NaN     NaN        NaN  
187            ALBANY        NY    12205      NaN     NaN        NaN  
224   KING OF PRUSSIA        PA    19406      NaN     NaN        NaN  
508        SOUTH BEND        IN    46635      NaN     NaN        NaN  
509      MERRILLVILLE        IN    46410      NaN     NaN        NaN  
3245            BOISE        ID    83713      NaN     NaN        NaN  
3616         NEWBURGH        IN    47630      NaN     NaN        NaN  
3617     INDIANAPOLIS        IN    46268      NaN     NaN        NaN  
3662       EARTH CITY        MO    63045      NaN     NaN        NaN  
3772       FORT WAYNE        IN    46825      NaN     NaN        NaN  
3852           DAYTON        OH    45414      NaN     NaN        NaN  
3960       YOUNGSTOWN        OH    44509      NaN     NaN        NaN  
4086          WYOMING        MI    49519      NaN     NaN        NaN  
4205         PORTLAND        OR    97220      NaN     NaN        NaN  
4473            TEMPE        AZ    85282      NaN     NaN        NaN  
4631   RANCHO CORDOVA    

### Observation: ITT is also listed several times in this file.
* Each entry has a unique school code, address, and zip code, suggesting that there is not one address or school code for each school, but each school has different identifiable information for each of its campuses.
* School Code may identify each individual campus of a school, and not each individual school.
* Example: 5610 and 5611 do not refer to different schools, but instead different campuses of the same school, ITT.

### Basic structural analysis:
* .describe(): Descriptive statistics
* .columns: Column names
* .shape: Shape of data
* .dtypes: Data types (such as object, int)
* .head(), tail(): First and last few rows of data
* sample(): Random sample of specified size
* .info(): Summary of data

In [16]:
fsl.describe()

ZipCode
count   6980.000000
mean   45657.029943
std    30989.290136
min        0.000000
25%    18930.250000
50%    44056.000000
75%    73415.750000
max    99801.000000

In [17]:
fsl.columns

Index(['SchoolCode', 'SchoolName', 'Address', 'City', 'StateCode', 'ZipCode',
       'Province', 'Country', 'PostalCode'],
      dtype='object')

In [18]:
fsl.shape

(6980, 9)

In [19]:
fsl.dtypes

SchoolCode    object
SchoolName    object
Address       object
City          object
StateCode     object
ZipCode        int64
Province      object
Country       object
PostalCode    object
dtype: object

#### Note: Zipcode is listed as an int

In [21]:
fsl.head(10)

SchoolCode                               SchoolName  \
0     B04724          WIDENER UNIV SCHOOL OF LAW - DE   
1     B06171      CENTER FOR ADVANCED STUDIES OF PUER   
2     B06511         PENTECOSTAL THEOLOGICAL SEMINARY   
3     B07022    THE CHICAGO SCHOOL OF PROF PSYCHOLOGY   
4     B07624     NATIONAL COLLEGE OF NATURAL MEDICINE   
5     B07625          OREGON COL OF ORIENTAL MEDICINE   
6     B08041             ALFRED ADLER GRADUATE SCHOOL   
7     B08083  UNIV OF THE DIST OF COLU -SCHOOL OF LAW   
8     B42154                 GRACE SCHOOL OF THEOLOGY   
9     E00014       AMRCN REPERTORY THTR INST ADV THTR   

                          Address        City StateCode  ZipCode Province  \
0   4601 CONCORD PIKE/PO BOX 7474  WILMINGTON        DE    19803      NaN   
1                      BOX S-4467    SAN JUAN        PR      902      NaN   
2                     PO BOX 3330   CLEVELAND        TN    37320      NaN   
3          325 NORTH WELLS STREET     CHICAGO        IL    60610      NaN   
4                   049 SW PORTER    PORTLAND        OR    97201      NaN   
5      10525 SE CHERRY BLOSSOM DR    PORTLAND        OR    97216      NaN   
6  1001 WEST HIGHWAY 7  SUITE 344     HOPKINS        MN    55305      NaN   
7      4200 CONNECTICUT AVENUE NW  WASHINGTON        DC    20008      NaN   
8            3705 COLLEGE PARK DR      CONROE        TX    77384      NaN   
9               64 BRATTLE STREET   CAMBRIDGE        MA     2138      NaN   

  Country PostalCode  
0     NaN        NaN  
1     NaN        NaN  
2     NaN        NaN  
3     NaN        NaN  
4     NaN        NaN  
5     NaN        NaN  
6     NaN        NaN  
7     NaN        NaN  
8     NaN        NaN  
9     NaN        NaN

In [31]:
fsl.tail(10)

SchoolCode                                SchoolName  \
6970     042515               BETH MEDRASH OF ASBURY PARK   
6971     042540             ULTIMATE TOUCH BARBER COLLEGE   
6972     042519                   FERRARA'S BEAUTY SCHOOL   
6973     042335                        MED-LIFE INSTITUTE   
6974     042503  HAIR ACADEMY SCHOOL OF BARBERING & BEAUT   
6975     042517         HOPE COLLEGE OF ARTS AND SCIENCES   
6976     E40419  BEAUTY INSTITUTE SCHWARZKOPF PROFESSIONA   
6977     042205  BUTTE COUNTY REGIONAL OCCUPATIONAL PROGR   
6978     G42404  UNIVERSIDAD ANA G. MENDEZ - CAMPUS VIRTU   
6979     042467          LEARNING BRIDGE CAREER INSTITUTE   

                          Address           City StateCode  ZipCode Province  \
6970          1500 VERMONT AVENUE       LAKEWOOD        NJ     8701      NaN   
6971         3732 SAUK TRAIL ROAD   RICHTON PARK        IL    60471      NaN   
6972      108-22 QUEENS BOULEVARD   FOREST HILLS        NY    11375      NaN   
6973      4103 EAST TAMIAMI TRAIL         NAPLES        FL    34112      NaN   
6974    1013 SOUTH COLLEGE AVENUE         NEWARK        DE    19713      NaN   
6975   1200 SOUTH WEST 3RD STREET  POMPANO BEACH        FL    33069      NaN   
6976         1411 RAILROAD AVENUE     BELLINGHAM        WA    98225      NaN   
6977        2491 CARMICHAEL DRIVE          CHICO        CA    95928      NaN   
6978    1552 AVENUE PONCE DE LEON       SAN JUAN        PR      926      NaN   
6979  1340 WEST TUNNELL BOULEVARD          HOUMA        LA    70360      NaN   

     Country PostalCode  
6970     NaN        NaN  
6971     NaN        NaN  
6972     NaN        NaN  
6973     NaN        NaN  
6974     NaN        NaN  
6975     NaN        NaN  
6976     NaN        NaN  
6977     NaN        NaN  
6978     NaN        NaN  
6979     NaN        NaN

In [32]:
fsl.sample(5)

SchoolCode                                SchoolName  \
5375     025875                 UNIVERSIDAD METROPOLITANA   
3498     006574       ST LUKE'S HOSPITAL OF BETHLEHEM, PA   
2828     003447             SPARTANBURG METHODIST COLLEGE   
4913     015873  COMMONWEALTH OF PR DEPT OF EDUC - ITPR R   
4351     013003                    NEILSON BEAUTY COLLEGE   

                    Address         City StateCode  ZipCode Province Country  \
5375           PO BOX 21150  RÍO PIEDRAS        PR      928      NaN     NaN   
3498          801 OSTRUM ST    BETHLEHEM        PA    18015      NaN     NaN   
2828  1000 POWELL MILL ROAD  SPARTANBURG        SC    29301      NaN     NaN   
4913   POST OFFICE BOX 7284        PONCE        PR      732      NaN     NaN   
4351       416 W. JEFFERSON       DALLAS        TX    75208      NaN     NaN   

     PostalCode  
5375        NaN  
3498        NaN  
2828        NaN  
4913        NaN  
4351        NaN

In [35]:
fsl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6980 entries, 0 to 6979
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   SchoolCode  6980 non-null   object
 1   SchoolName  6980 non-null   object
 2   Address     6980 non-null   object
 3   City        6980 non-null   object
 4   StateCode   6980 non-null   object
 5   ZipCode     6980 non-null   int64 
 6   Province    111 non-null    object
 7   Country     401 non-null    object
 8   PostalCode  227 non-null    object
dtypes: int64(1), object(8)
memory usage: 490.9+ KB


# Manually import sheets
* Import each sheet from the original Excel file as its own pandas dataframe
* Print the first 6 rows of each dataframe
* Reason: Each sheet was originally loaded into a dictionary of dataframes. Loading each sheet as its own individual dataframe will make datacleaning easier.

In [28]:
qa_raw = pd.read_excel("FL_Dashboard_AY2009_2010_Q1.xls", sheet_name="Quarterly Activity")
ays_raw = pd.read_excel("FL_Dashboard_AY2009_2010_Q1.xls", sheet_name="Award Year Summary")

### Quarterly Analysis

In [29]:
qa_raw.head(6)

2009-2010 Award Year FFEL Volume by School  \
0  Award Year Quarterly Activity  (07/01/2009-09/...   
1                                 Data Run: 4/5/2012   
2                                                      
3                                                NaN   
4                                             OPE ID   
5                                           00106100   

                  Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4  \
0                        NaN        NaN        NaN          NaN   
1                        NaN        NaN        NaN          NaN   
2                        NaN        NaN        NaN          NaN   
3                        NaN        NaN        NaN          NaN   
4                     School      State   Zip Code  School Type   
5  ALASKA PACIFIC UNIVERSITY         AK  995084672      PRIVATE   

        Unnamed: 5             Unnamed: 6             Unnamed: 7  \
0              NaN                    NaN                    NaN   
1              NaN                    NaN                    NaN   
2              NaN                    NaN                    NaN   
3  FFEL SUBSIDIZED                    NaN                    NaN   
4       Recipients  # of Loans Originated  $ of Loans Originated   
5              291                    291                1546994   

           Unnamed: 8          Unnamed: 9  ...       Unnamed: 15  \
0                 NaN                 NaN  ...               NaN   
1                 NaN                 NaN  ...               NaN   
2                 NaN                 NaN  ...               NaN   
3                 NaN                 NaN  ...  FFEL PARENT PLUS   
4  # of Disbursements  $ of Disbursements  ...        Recipients   
5                 292              830513  ...                31   

             Unnamed: 16            Unnamed: 17         Unnamed: 18  \
0                    NaN                    NaN                 NaN   
1                    NaN                    NaN                 NaN   
2                    NaN                    NaN                 NaN   
3                    NaN                    NaN                 NaN   
4  # of Loans Originated  $ of Loans Originated  # of Disbursements   
5                     33                 386770                  35   

          Unnamed: 19     Unnamed: 20            Unnamed: 21  \
0                 NaN             NaN                    NaN   
1                 NaN             NaN                    NaN   
2                 NaN             NaN                    NaN   
3                 NaN  FFEL GRAD PLUS                    NaN   
4  $ of Disbursements      Recipients  # of Loans Originated   
5              192181               5                      5   

             Unnamed: 22         Unnamed: 23         Unnamed: 24  
0                    NaN                 NaN                 NaN  
1                    NaN                 NaN                 NaN  
2                    NaN                 NaN                 NaN  
3                    NaN                 NaN                 NaN  
4  $ of Loans Originated  # of Disbursements  $ of Disbursements  
5                  69730                   5               34865  

[6 rows x 25 columns]

### Award Year Summary

In [30]:
ays_raw.head(6)

2009-2010 Award Year FFEL Volume by School  \
0  Award Year Cumulative Activity through Quarter...   
1                                 Data Run: 4/5/2012   
2                                                      
3                                                NaN   
4                                             OPE ID   
5                                           00106100   

                  Unnamed: 1 Unnamed: 2 Unnamed: 3   Unnamed: 4  \
0                        NaN        NaN        NaN          NaN   
1                        NaN        NaN        NaN          NaN   
2                        NaN        NaN        NaN          NaN   
3                        NaN        NaN        NaN          NaN   
4                     School      State   Zip Code  School Type   
5  ALASKA PACIFIC UNIVERSITY         AK  995084672      PRIVATE   

        Unnamed: 5             Unnamed: 6             Unnamed: 7  \
0              NaN                    NaN                    NaN   
1              NaN                    NaN                    NaN   
2              NaN                    NaN                    NaN   
3  FFEL SUBSIDIZED                    NaN                    NaN   
4       Recipients  # of Loans Originated  $ of Loans Originated   
5              291                    291                1546994   

           Unnamed: 8          Unnamed: 9  ...       Unnamed: 15  \
0                 NaN                 NaN  ...               NaN   
1                 NaN                 NaN  ...               NaN   
2                 NaN                 NaN  ...               NaN   
3                 NaN                 NaN  ...  FFEL PARENT PLUS   
4  # of Disbursements  $ of Disbursements  ...        Recipients   
5                 292              830513  ...                31   

             Unnamed: 16            Unnamed: 17         Unnamed: 18  \
0                    NaN                    NaN                 NaN   
1                    NaN                    NaN                 NaN   
2                    NaN                    NaN                 NaN   
3                    NaN                    NaN                 NaN   
4  # of Loans Originated  $ of Loans Originated  # of Disbursements   
5                     33                 386770                  35   

          Unnamed: 19     Unnamed: 20            Unnamed: 21  \
0                 NaN             NaN                    NaN   
1                 NaN             NaN                    NaN   
2                 NaN             NaN                    NaN   
3                 NaN  FFEL GRAD PLUS                    NaN   
4  $ of Disbursements      Recipients  # of Loans Originated   
5              192181               5                      5   

             Unnamed: 22         Unnamed: 23         Unnamed: 24  
0                    NaN                 NaN                 NaN  
1                    NaN                 NaN                 NaN  
2                    NaN                 NaN                 NaN  
3                    NaN                 NaN                 NaN  
4  $ of Loans Originated  # of Disbursements  $ of Disbursements  
5                  69730                   5               34865  

[6 rows x 25 columns]

# <u>Overall Observations:</u>
### Our original questions...
1. Do community colleges or technical colleges originate more loans overall?
2. Which states have the highest total loan disbursements?
3. Do private or public schools disburse more in federal student loans?
4. Do institutions with “college” in the name receive more loans than those with “university”?
5. Are unsubsidized loans more common at private colleges than public ones?


# Quarterly Analysis and Award Year Summary
* Missing values as well as "0s" will need to be handled during data cleaning
* The first few missing rows will need to be handled (dropped)
* Columns will need to be renamed to proper original column names (School, State, etc.)
* Column headers (FFEL SUBSIDIZED, FFEL PARENT PLUS) will need to be reintegrated into the data. They currently exist among empty rows.

# Federal School List (School Codes)
* Each entry has a unique school code, address, and zip code, suggesting that there is not one address or school code for each school, but each school has different identifiable information for each of its campuses.
* School Code may identify each individual campus of a school, and not each individual school.
* Example: 5610 and 5611 do not refer to different schools, but instead different campuses of the same school, ITT.
* Schools may need to be grouped by name instead of state, as one school contains several school codes across several states.